lib/web_ui/lib/src/engine/embedder.dart

```diff
@@ -202,9 +207,10 @@ class FlutterViewEmbedder {
       // elements transparent. This way, if a platform view appears among other
       // interactive Flutter widgets, as long as those widgets do not intersect
       // with the platform view, the platform view will be reachable.
-      semanticsHostElement,
```

Comment:
@yjbanov would love to get your thoughts on if we can safely move the semantics tree outside of the shadowDOM and into its own node?

Reply:
With the current problematic implementation of semantics, this shouldn't break anything I know of. However, one issue with the current semantics tree is that it does not contain platform views. This creates some issues with traversal order. I was thinking in the future to slice the semantics tree and interleave it with the render tree such that platform views can be embedded into the semantics tree while preserving the paint order. To do that, the semantics tree would have to be inside the shadow root, otherwise the `<slot>` elements won't work.

Having said that, I'm willing to cross that bridge when we get there. What's the motivation for moving it out?

Reply:
Primary motivation to move it out is to fix the password autofill issue for when semantics is enabled, since `<input>`s will be rendered in the semantics tree in that case and they'll face the same shadowDOM issues.  AFAIK, the 2 nodes that host text editing elements would be whatever host we specify in `text_editing.dart` and the semantics tree.